src/main/scala/de/heikoseeberger/sbtheader/HeaderPlugin.scala

```diff
@@ -127,4 +137,23 @@ object HeaderPlugin extends AutoPlugin {
     log.debug(s"About to create/update header for $file")
     HeaderCreator(headerPattern, headerText, log, new FileInputStream(file)).getText.map(write)
   }
+
+  private def checkHeadersTask(files: Seq[File], headers: Map[String, (Regex, String)], log: Logger) = {
+    val filesWithoutHeader = files
+      .groupBy(_.extension)
+      .collect { case (Some(ext), groupedFiles) => headers.get(ext).map(_ -> groupedFiles) }
+      .flatten
+      .flatMap { case ((pattern, text), groupedFiles) => groupedFiles.flatMap(checkHeader(pattern, text, log)) }
```

Comment:
Structurally redundant with the definition of `createHeaders`. I didn't find a way to better abstract the commons parts. Ideas?